app/models/tenant.rb

```diff
@@ -317,6 +318,10 @@ def create_miq_product_features_for_tenant_nodes
     MiqProductFeature.seed_single_tenant_miq_product_features(self)
   end
 
+  def destroy_with_subtree
+    subtree.sort_by { |d| -d.depth }.each(&:destroy)
```

Comment:
for some reason this is a little easier for me to read. but don't have good reason for why you should change it

But I'm hoping that ancestry will be able to destroy these for us.

```suggestion
    subtree.sort_by(&:depth).reverse.each(&:destroy)
```